android/fakewallet/src/main/java/com/solana/mobilewalletadapter/fakewallet/MobileWalletAdapterActivity.kt

```diff
@@ -31,6 +32,23 @@ class MobileWalletAdapterActivity : AppCompatActivity() {
             }
         }
 
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.CREATED) {
+                viewModel.lowPower.collect {
+                    // should use dialog fragment, etc. but this is a quick demo
+                    AlertDialog.Builder(this@MobileWalletAdapterActivity)
+                        .setTitle(R.string.low_power_mode_warning_title)
+                        .setMessage(R.string.str_low_power_mode_warning_dsc)
+                        .setPositiveButton(R.string.label_return_to_dapp) { _, _ ->
+                            Log.w(TAG, "Connection failed due to device low power mode, returning to dapp.")
+                            finish()
+                        }
+                        .setNeutralButton(R.string.label_close_dialog) { d, _ -> d.dismiss() }
```

Comment:
Do we really want a neutral button here? Real wallets can be more complex if they desire, but I'd propose we only implement the straightforward case for `fakewallet`, which in this instance is the positive button you've defined above.